src/field_theory/galois.lean

```diff
@@ -0,0 +1,233 @@
+
+theorem fixing_subgroup_of_fixed_field [finite_dimensional F E] :
+  fixing_subgroup (fixed_field H) = H :=
+begin
+  have H_le : H ≤ (fixing_subgroup (fixed_field H)) := (le_iff_le _ _).mp (le_refl _),
+  suffices : fintype.card H = fintype.card (fixing_subgroup (fixed_field H)),
+  { exact subgroup.ext' (set.eq_of_inclusion_surjective ((fintype.bijective_iff_injective_and_card
+    (set.inclusion H_le)).mpr ⟨set.inclusion_injective H_le, this⟩).2).symm },
+  rw fintype.card_congr (fixing_subgroup_iso (fixed_field H)).to_equiv,
+  rw fintype.card_congr (fixed_points.to_alg_hom_equiv H E),
+  rw fintype.card_congr (alg_equiv_equiv_alg_hom (fixed_field H) E),
+  exact fintype.card_congr (by refl),
+end
+
+instance alg_instance : algebra K (fixed_field (fixing_subgroup K)) :=
+{ smul := λ x y, ⟨x*y, λ ϕ, by rw [smul_mul', (show ϕ • ↑x = ↑x, by exact subtype.mem ϕ x),
+    (show ϕ • ↑y = ↑y, by exact subtype.mem y ϕ)]⟩,
+  to_fun := λ x, ⟨x, λ ϕ, subtype.mem ϕ x⟩,
+  map_zero' := rfl,
+  map_add' := λ _ _, rfl,
+  map_one' := rfl,
+  map_mul' := λ _ _, rfl,
+  commutes' := λ _ _, mul_comm _ _,
+  smul_def' := λ _ _, rfl }
+
+instance tower_instance : is_scalar_tower K (fixed_field (fixing_subgroup K)) E :=
+⟨λ _ _ _, mul_assoc _ _ _⟩
+
+theorem fixed_field_of_fixing_subgroup [finite_dimensional F E] [h : is_galois F E] :
+  fixed_field (fixing_subgroup K) = K :=
+begin
+  have K_le : K ≤ fixed_field (fixing_subgroup K) := (le_iff_le _ _).mpr (le_refl _),
+  suffices : findim K E = findim (fixed_field (fixing_subgroup K)) E,
+  { exact (intermediate_field.eq_of_le_of_findim_eq' K_le this).symm },
+  rw [findim_fixed_field_eq_card, fintype.card_congr (fixing_subgroup_iso K).to_equiv],
+  exact (is_galois_implies_card_aut_eq_findim K E).symm,
+end
+
+lemma card_fixing_subgroup_eq_findim [finite_dimensional F E] [is_galois F E] :
+  fintype.card (fixing_subgroup K) = findim K E :=
+by conv { to_rhs, rw [←fixed_field_of_fixing_subgroup K, findim_fixed_field_eq_card] }
+
+/-- The Galois correspondence from intermediate fields to subgroups -/
+def galois_correspondence [finite_dimensional F E] [is_galois F E] :
+  intermediate_field F E ≃o order_dual (subgroup (E ≃ₐ[F] E)) :=
+{ to_fun := fixing_subgroup,
+  inv_fun := fixed_field,
+  left_inv := λ K, fixed_field_of_fixing_subgroup K,
+  right_inv := λ H, fixing_subgroup_of_fixed_field H,
+  map_rel_iff' := λ K L, by { rw [←fixed_field_of_fixing_subgroup L, le_iff_le,
+                                  fixed_field_of_fixing_subgroup L, ←order_dual.dual_le], refl} }
+
+end galois_correspondence
+
+section galois_equivalent_definitions
+
+variables (F : Type*) [field F] (E : Type*) [field E] [algebra F E]
+
+lemma is_separable_splitting_field_of_is_galois [finite_dimensional F E] [h : is_galois F E] :
+  ∃ p : polynomial F, p.separable ∧ p.is_splitting_field F E :=
+begin
+  cases field.exists_primitive_element h.1 with α h1,
+  cases h.1 α with h2 h3,
+  cases h.2 α with _ h4,
```

Comment:
I see this pattern come up a few times. Maybe define `is_galois.is_integral` `is_galois.separable` and `is_galois.splits` that return the `h2`, `h3` and `h4` here respectively?